capstone/src/main/webapp/js/calendar.js

```diff
@@ -0,0 +1,78 @@
+const script = document.createElement('script');
+script.src = 'https://www.gstatic.com/charts/loader.js';
+script.defer = true;
+
+window.initMap = function() {
+  google.charts.load('current', {'packages': ['calendar', 'timeline']});
+  google.charts.setOnLoadCallback(drawCharts);
+
+  /**
+   * Callback function to generate charts.
+   */
+  function drawCharts() {
+    const freqData = new google.visualization.DataTable();
+    const timelineData = new google.visualization.DataTable();
+    fetch('/suggested-cal').
+        then((response) => response.json()).
+        then((deadlineInfo) =>
+          populateDataTables(freqData, timelineData, deadlineInfo));
+    drawFrequencyChart(freqData);
+    drawTimelineChart(timelineData);
```

Comment:
Should this also be part of a then() so make sure the data has been populated on the data tables? Since fetch() runs asynchronously.

Reply:
Will move those function calls into a `then()`.